Frameworks/CoreGraphics/CGGraphicBufferImage.mm

```diff
@@ -30,9 +40,124 @@
 
 #pragma clang diagnostic pop
 
+using namespace Microsoft::WRL;
+
 static const wchar_t* TAG = L"CGGraphicBufferImage";
 extern int imgDataCount;
 
+/**
+ * Custom Core Graphic IWICBitmap interface that uses the backing image buffer without copying.
+ */
+
+class CGGraphicBufferWICBitmap;
+
+class CGGraphicBufferWICBitmapLock
+    : public RuntimeClass<RuntimeClassFlags<RuntimeClassType::ClassicCom>, IAgileObject, FtmBase, IWICBitmapLock> {
+public:
+    CGGraphicBufferWICBitmapLock(
+        _In_ UINT width, _In_ UINT height, _In_ UINT stride, _In_ WICPixelFormatGUID pixelFormat, _In_ BYTE* dataBuffer)
+        : m_width(width), m_height(height), m_stride(stride), m_pixelFormat(pixelFormat), m_dataBuffer(dataBuffer) {
+    }
+
+    HRESULT STDMETHODCALLTYPE GetSize(_Out_ UINT* width, _Out_ UINT* height) {
+        *width = m_width;
+        *height = m_height;
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetStride(_Out_ UINT* stride) {
+        *stride = m_stride;
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetDataPointer(_Out_ UINT* bufferSize, _Outptr_ WICInProcPointer* data) {
+        *bufferSize = m_height * m_stride;
+        *data = m_dataBuffer;
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetPixelFormat(_Out_ WICPixelFormatGUID* pixelFormat) {
+        *pixelFormat = m_pixelFormat;
+        return S_OK;
+    }
+
+private:
+    UINT m_width;
+    UINT m_height;
+    UINT m_stride;
+    WICPixelFormatGUID m_pixelFormat;
+    BYTE* m_dataBuffer;
+};
+
+class CGGraphicBufferWICBitmap : public RuntimeClass<RuntimeClassFlags<RuntimeClassType::ClassicCom>, IAgileObject, FtmBase, IWICBitmap> {
+public:
+    CGGraphicBufferWICBitmap(_In_ CGGraphicBufferImageBacking* imageBacking) : m_imageBacking(imageBacking) {
+    }
+
+    ~CGGraphicBufferWICBitmap() {
+        m_imageBacking->ReleaseImageData();
```

Comment:
>m_imageBacking [](start = 8, length = 14)

This is not correct - you should pass the image backing to the lock object and ReleaseImageData when the lock object is released.  Otherwise, we are locking too many times (everytime D2D locks and not release enough times).